plugins/aws-appmesh/plugin/commands.go

```diff
@@ -302,8 +323,27 @@ func (plugin *Plugin) deleteIngressRules(
 
 // deleteEgressRules deletes the iptable rules for egress traffic.
 func (plugin *Plugin) deleteEgressRules(iptable *iptables.IPTables) error {
+
+	// Check if egress chain exist before deleting
+	exists := false
+	chains, err := iptable.ListChains("nat")
+	if err != nil {
+		return log.Errorf("failed to list chains: %v", err)
+	}
+	for _, ch := range chains {
+		if ch == egressChain {
+			exists = true
+			break
+		}
+	}
+	//return nil to make sure deleting process is idempotent
+	if !exists {
+		log.Warn("deleting a non-existent egress chain: %s", egressChain)
+		return nil
+	}
```

Comment:
Not necessary, but I'd be tempted to create a standalone function that checks whether a given chain exists, then use it for both the egress and ingress check.

Reply:
Nice call, will do.